### ISSUE TITLE
LVGL9 cannot output Log

### ISSUE THREAD

**Issue by arilink** [lvgl/lv_port_pc_vscode#37]
I use lv_port_pc_eclipse and lv_port_pc_vscode to add Printf or LV_LOG_ERROR and I can't output log. Why?

**Comment by arilink** [comment]
@kisvegabor  @microwavesafe  @SIvaCoHan @rzr

**Comment by kisvegabor** [comment]
I see it on the Terminal tab:
![image](https://github.com/lvgl/lv_port_pc_vscode/assets/7599318/5e9dbf69-3ac6-4176-a281-f4c93048ece3)

**Comment by arilink** [comment]
Are you the latest code pulled? I can't print using printf or LV_LOG_Error

**Comment by kisvegabor** [comment]
Yes, I tried the latest with printf. Do you check the terminal tab?

**Comment by arilink** [comment]
yes.  no log show

**Comment by arilink** [comment]
![image](https://github.com/lvgl/lv_port_pc_vscode/assets/97713657/2ce2a68a-2c71-4729-9a00-be2ac2c513c3)

**Comment by kisvegabor** [comment]
What is `format:`? Can you try just a simply `printf("Hello\n");`

**Comment by arilink** [comment]
it is just a  code hint

**Comment by arilink** [comment]
I'll just use printf("Hello\n"); The same

**Comment by arilink** [comment]
> What is `format:`? Can you try just a simply `printf("Hello\n");`

Can you shared your Code?  I use it to compare

**Comment by kisvegabor** [comment]
Really nothing special. Just cloned the project as it is and added `printf("asd\n")` in `main.c`.

**Comment by arilink** [comment]
@kisvegabor   Hello senior, I re-built a V8.4 development environment using CLion, which can output log normally. Then I used CLion_V8.4 code to replace the Driver and LVGL code in LV_For_VS_Code respectively, which still cannot output Log, so it can be inferred that, There is a problem with my computer or LV_For_VS_Code code. Could you please help send an LV_For_VS_Code Project in which you can output Log so that I can troubleshoot the computer problem

**Comment by kisvegabor** [comment]
That's strange, however I'm using this repo as it is, so you can just clone my project from here :slightly_smiling_face: 

Maybe @danyyliu has some ideas.

**Comment by arilink** [comment]
<img width="1470" alt="bbc51fdfdabf8034aad1625969a70da" src="https://github.com/lvgl/lv_port_pc_vscode/assets/97713657/f0fa30c2-39a4-48ad-8668-b95208553cbe">
I try to use uart to show log. it still no log...

**Comment by arilink** [comment]
yes  use uart log ok，but  use default still not work，should i enable  LV_OS_WINDOWS？

**Comment by arilink** [comment]
> So it's an UART log. I completely missed that part so far.
> 
> I think the problem is that you need to disable `LV_LOG_PRINTF`. If it's enabled the custom print callback is not called.

No. I fan't fix this problem. so i  try to use windows uart

**Comment by arilink** [comment]
I find a new way to fix this problem.but when i use LV_LOG_XX. the color log is not work.   The specific content is as follows：
void use_console_output()        //打印窗口
{
    AllocConsole();
    FILE *stream;
    freopen_s(&stream, "CONOUT$", "w", stdout);
    freopen_s(&stream, "CONIN$", "r", stdin);
}

![image](https://github.com/lvgl/lv_port_pc_vscode/assets/97713657/e560b55f-a4d7-43ee-b5ca-a8dedc46eb90)

**Comment by danyyliu** [comment]
In my mind, this is an issue caused by the Windows system.
![image](https://github.com/lvgl/lv_port_pc_vscode/assets/24490390/7739b032-b673-48b4-a638-0e4b81c41f27)

From the previous information, I think we can get a conclusion that printf doesn't work, while uart is ok.
It doesn't make sense that printf doesn't work, unless there are some code is manipulating the console, which can also explain why the issue can be fixed by calling AllocConsole.
I don't think LVGL can support color log for windows console now, since SetConsoleTextAttribute should be used to set the color.

I'm not an expert of Windows program, please let me know if I am correct.

**Comment by arilink** [comment]
So from the current results, the problem is not related to LVGL, which has the log ready, but the windows console is not starting correctly or vscode is not receiving information from the console

**Comment by arilink** [comment]
<img width="838" alt="a3b20db812bb1e73bf27ce0773513da" src="https://github.com/lvgl/lv_port_pc_vscode/assets/97713657/3827730e-d0e9-4540-b4ce-1e28766f12a0">
<img width="846" alt="ed41dd7f4e91f171e7aef04957ca33a" src="https://github.com/lvgl/lv_port_pc_vscode/assets/97713657/b10e909f-5ba6-44eb-a2c7-e46c4eab1e9b">

<img width="842" alt="7aa5248e86fa514c3460859be1c3244" src="https://github.com/lvgl/lv_port_pc_vscode/assets/97713657/bef71163-4124-4f7d-b201-55303f56c63d">

<img width="844" alt="dde9df557c4ece963e9edad112f0893" src="https://github.com/lvgl/lv_port_pc_vscode/assets/97713657/472699ec-f2a0-440f-ac7b-7a491724a02a">
I use CLion for comparison. When CLion runs the demo, it will open a new console, and the information in the console will be displayed in the IDE, while VSCode does not. Does this mean that VSCode did not wake up the console correctly?
@kisvegabor  @danyyliu  Can you see how many consoles are added to the system when you run the demo using VSCode?

**Comment by danyyliu** [comment]
I haven't done any test like this in Windows. Could you please try it with hello world test without LVGL? I want to see what happens with the console in this situation.

**Comment by arilink** [comment]
![image](https://github.com/lvgl/lv_port_pc_vscode/assets/97713657/ebe7d8c7-e6e7-49cd-bc11-f36939611220)
It can print correct log

**Comment by danyyliu** [comment]
So, it should not be an issue of vscode or clion. Maybe you can check if there is any code in your project is doing something with the console.

I tried to build and run this project in Windows, it seems the console can work well:
![image](https://github.com/lvgl/lv_port_pc_vscode/assets/24490390/34700320-2fe5-4739-ad48-03f66a4ae508)

**Comment by arilink** [comment]
This is the only project I opened and I didn't do anything else with the console, which is weird

**Comment by arilink** [comment]
![image](https://github.com/lvgl/lv_port_pc_vscode/assets/97713657/4acbd475-e928-4d9b-8325-819865ed7863)

**Comment by arilink** [comment]
@danyyliu  I suddenly thought of a point, is it related to SDL_Main?
![image](https://github.com/lvgl/lv_port_pc_vscode/assets/97713657/645cf6b0-ec9a-405a-a036-6071cd276a70)

**Comment by vgegok** [comment]
![image](https://github.com/lvgl/lv_port_pc_vscode/assets/45798060/c833d9b0-d533-497d-a228-4b1d65785b65)
![image](https://github.com/lvgl/lv_port_pc_vscode/assets/45798060/7e4d49e9-5be3-4dd2-9994-9580b1b531f0)

I think this is related to SDL.

**Comment by kisvegabor** [comment]
Does simple printf work instead of LV_LOG?

**Comment by vgegok** [comment]
I tested MacOS and everything is normal.

**Comment by kisvegabor** [comment]
How does your `lv_conf.h` looks like? Please also open `lvgl` to the latest from `master`. There some logging related fixes earlier.

**Comment by AndreCostaaa** [comment]
Hi everyone

Are you on windows @FGQ1001 ? Is your `lv_conf.h` the default one provided by this repository ?
Could you confirm that everything works in your shell i.e:

```bash
cmake -B build -S .
cmake --build build
.\bin\main.exe
[Warn]	(0.000, +0)	lv_init: Memory integrity checks are enabled via LV_USE_ASSERT_MEM_INTEGRITY which makes LVGL much slower lv_init.c:309
[Warn]	(0.000, +0)	lv_init: Object sanity checks are enabled via LV_USE_ASSERT_OBJ which makes LVGL much slower lv_init.c:313
[Warn]	(0.000, +0)	lv_init: Style sanity checks are enabled that uses more RAM lv_init.c:317
```

These warns should appear by default with the default config

**Comment by YobeZhou** [comment]
fix: output logs on platform Window https://github.com/lvgl/lv_port_pc_vscode/pull/84/commits/93379065440405f4a681a7b87d6281739f92d8ca

**Comment by AndreCostaaa** [comment]
I believe we can now close this issue. Let me know if you encounter any issues

Thank you @YobeZhou !
